helmutils/helm_indent_finder_test.go

```diff
@@ -0,0 +1,244 @@
+package helmutils_test
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/solo-io/go-utils/helmutils"
+)
+
+var _ = Describe("helmutils indent finder", func() {
+	var opts helmutils.HelmDetectOptions
+	Context("should pass on the following", func() {
+		It("should not detect fine yaml", func() {
+			data := `
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: gloo
+    gloo: rate-limit
+spec:
+`
+			badWindows := helmutils.FindHelmChartWhiteSpaces(data, opts)
+			Expect(len(badWindows)).To(Equal(0))
+		})
+		It("should pass with a comment line", func() {
+			data := `
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+              # app: gloo
+    gloo: rate-limit
+spec:
+`
+			badWindows := helmutils.FindHelmChartWhiteSpaces(data, opts)
+			Expect(len(badWindows)).To(Equal(0))
+		})
+		It("should not detect empty lines", func() {
+			data := `
+
+`
+			badWindows := helmutils.FindHelmChartWhiteSpaces(data, opts)
+			Expect(len(badWindows)).To(Equal(0))
+		})
+		It("should not detect empty lines with spaces", func() {
+			data := `
+        
+`
+			badWindows := helmutils.FindHelmChartWhiteSpaces(data, opts)
+			Expect(len(badWindows)).To(Equal(0))
+		})
+	})
+
+	Describe("dealing with white spaces in helmutils", func() {
+		It("should be able to detect 4 spaces, when it should be 2", func() {
+			data := `
+apiVersion: v1
+kind: Service
+metadata:
+   labels:
+    app: gloo
+    gloo: rate-limit
+  name: rate-limit
+  namespace: default
+spec:
+`
+			badWindows := helmutils.FindHelmChartWhiteSpaces(data, opts)
+			Expect(len(badWindows)).To(Equal(2))
```

Comment:
would it be worth to also do an Expect that the array of badWindows are the windows we expect?

Reply:
don't think so.
Since the window size is relatively large compared to the input.
But I guess the statement should be, should we output the line that causes the issue, this can be up for debate.  We could return a structure that contains metadata about these windows.   

The reason I chose windows is that it was hard to identify the area in the code that was causing the problem. Especially if the code was re-used multiple times.